api-authorization/server/lib/authorization-middleware.ts

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { NextFunction, Request, Response } from 'express';
+import jwt, { JwtPayload } from 'jsonwebtoken';
+import { ClientError } from './client-error.js';
+
+interface PayLoad extends JwtPayload {
+  userId: number;
+}
+
+export function authMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  const auth = req.get('authorization');
+  const token = auth?.split('Bearer ')[1];
+  const secretToken = process.env.TOKEN_SECRET;
+  if (!secretToken) throw new ClientError(401, 'Auth failed');
+  if (!auth || !token) {
+    throw new ClientError(401, 'Auth required');
+  }
+  jwt.verify(token, secretToken, (err, decoded) => {
```

Comment:
Don't use the callback version. Just get the returned value from `jwt.verify()` and use it.

Reply:
Wow, this revision really made the code look so much prettier, thank you!